File: 2128-reverse-prefix-of-word/2128-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int idx = word.indexOf(ch);
        if(idx == -1)
            return word;
        StringBuilder res = new StringBuilder(word.substring(0, idx + 1));
        res.reverse();
        res.append(word.substring(idx + 1));
        return res.toString();
    }
}

File: 2128-reverse-prefix-of-word/2128-reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        lst = list(word)
        flag = False
        for right in range(len(lst)):
            if lst[right] == ch:
                flag = True
                break
        print(right)
        if not flag:
            return ''.join(lst)
        if flag:
            left = 0
            while left <= right:
                lst[left], lst[right] = lst[right], lst[left]
                left+= 1
                right -= 1
        return ''.join(lst)